components/centraldashboard/public/components/main-page.js

```diff
@@ -360,19 +374,23 @@ export class MainPage extends utilitiesMixin(PolymerElement) {
         });
         if (hashPath) {
             matchPath = path + '#' + hashPath;
-            matchingLink = allLinks
-                .find((l) => this.compareLinks(l, matchPath));
+            this.matchingIndex = allLinks
+                .findIndex((l) => this.compareLinks(l, matchPath));
         } else {
```

Comment:
@kimwnasptd Good point ! When the namespace is changed by the namespace selector, both current URL and menu links are updated. If current URL is updated earlier, the new URL(e.g /mlflow/workspace1) can't match with the prefix of old menu links (e.g /mlflow/workspace2).

In that case, all `path.startsWith(link)` may be False, and matchingIndex is not updated. Making it class variable `this.matchingIndex`, UI can remember which menu item is selected, and re-link the URL and menu-bar.

Reply:
Makes sense, thanks for the explanation!